### PR TITLE
Added react components for authentication

### PR DESCRIPTION
**Pull request by prasanthchaduvula**
Closes 
https://www.notion.so/saeloun/Redevelop-Sign-In-page-using-React-1ccc0724feb24e7485a02bdf4c51a17a
https://www.notion.so/saeloun/Redevelop-Sign-Up-page-using-React-58ea46ef7bbd4b239b191ca1863d1063

### Why
- Currently we have authenitcaion views on rails views. We want to move our routing completely to react and use rails API to handle requests.

### What
- This PR is part of feature:move-authentication to react. In this PR we created react components to handle all the authentication views like Signup, Signin, ForgotPassword and ResetPassword

### CODE REVIEW

**Comment by prasanthchaduvula** [comment]
@vipulnsward checks are failing for the below reason. I tried rebuilding them but didn't work. Please review it. This PR only included React components.
Screenshot 2023-02-22 at 3 02 46 PM

File: app/javascript/src/components/Authentication/FeaturePreviews/utils.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  MiruManagingEmployeesScreen,
+} from "miruIcons";
+
+export const carouselItems = [
+  {
+    image: MiruTimeTrackingScreen,
+    texts: {
+      title: "Start with effortless time tracking today",
+      description:
+        "Get a detailed report of your team's efforts & measure their time distribution across different projects. Identify team burnouts & underworked members for better decision-making.",
```
Comment (vipulnsward) [inline app/javascript/src/components/Authentication/FeaturePreviews/utils.ts:16]: In future, lets move these to I18n cc @supriya3105

File: app/javascript/src/components/Authentication/ResetPassword/utils.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export const resetPasswordFormInitialValues = {
+  password: "",
+  confirm_password: "",
+};
+
+export const resetPasswordFormValidationSchema = Yup.object().shape({
+  password: Yup.string()
+    .matches(
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})/, // eslint-disable-line
```
Comment (vipulnsward) [inline app/javascript/src/components/Authentication/ResetPassword/utils.ts:11]: Can we try using https://www.npmjs.com/package/owasp-password-strength-test ?

Comment (prasanthchaduvula) [reply app/javascript/src/components/Authentication/ResetPassword/utils.ts:11]: Will look for how to achieve this with yup validation in my next PR

File: app/javascript/src/components/Authentication/SignIn/SignInForm.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                          )}
+                        </span>
+                      </div>
+                    </div>
+                    <div className="mb-6 block text-xs tracking-wider text-red-600">
+                      {errors.password && touched.password && (
+                        <div>{errors.password}</div>
+                      )}
+                    </div>
+                    {/* Sign In Button */}
```
Comment (vipulnsward) [inline app/javascript/src/components/Authentication/SignIn/SignInForm.tsx:126]: ✂️

File: app/javascript/src/components/Authentication/SignIn/SignInForm.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+                      <span className="mx-4 flex-shrink text-xs text-miru-dark-purple-1000">
+                        or
+                      </span>
+                      <div className="flex-grow border-t border-miru-gray-1000" />
+                    </div>
+                  </Form>
+                );
+              }}
+            </Formik>
+            {/* Sign In with Google */}
```
Comment (vipulnsward) [inline app/javascript/src/components/Authentication/SignIn/SignInForm.tsx:152]: ✂️

File: app/javascript/src/components/Authentication/SignIn/SignInForm.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+              <button
+                className="form__button whitespace-nowrap"
+                data-cy="sign-up-button"
+                onClick={authenticationApi.googleAuth}
+              >
+                <img alt="" className="mr-2" src={GoogleSVG} />
+                Sign In with Google
+              </button>
+            </div>
+            {/* Sign In with Apple */}
```
Comment (vipulnsward) [inline app/javascript/src/components/Authentication/SignIn/SignInForm.tsx:163]: ✂️

**Review by vipulnsward — APPROVED** [review]
@prasanthchaduvula some more comments, please merge on green. Also ask @apoorv1316 to verify once tomorrow

**Pull request merged**